frappe/public/js/frappe/form/controls/duration.js

```diff
@@ -0,0 +1,145 @@
+frappe.ui.form.ControlDuration = frappe.ui.form.ControlData.extend({
+	make_input: function() {
+		this._super();
+		this.make_picker();
+	},
+
+	make_picker: function() {
+		this.inputs = [];
+		this.set_duration_options();
+		this.$picker = $(
+			`<div class="duration-picker">
+				<div class="picker-row row"></div>
+			</div>`
+		);
+		this.$wrapper.append(this.$picker);
+		this.build_numeric_input("days", !this.duration_options.showDays);
+		this.build_numeric_input("hrs", false);
+		this.build_numeric_input("mins", false);
+		this.build_numeric_input("secs", !this.duration_options.showSeconds);
+		this.set_duration_picker();
+		this.$picker.hide();
+		this.bind_events();
+		this.refresh();
+	},
+
+	build_numeric_input: function(label, hidden, max) {
+		let $duration_input = $(`
+			<input class="input-sm duration-input" data-duration="${label}" type="number" min="0" value="0">
+		`)
+
+		let $input = $(`<div class="row duration-row"></div>`).prepend($duration_input);
+
+		if (max) {
+			$duration_input.attr("max", max);
+		}
+
+		this.inputs[label] = $duration_input;
+
+		let $control = $(`
+			<div class="col duration-col">
+				<div class="row duration-row duration-label">${label}</div>
```

Comment:
Translate the label here. Also add the labels as comments for the parser to pick